src/base/utils/compare.h

```diff
@@ -56,6 +58,11 @@ namespace Utils::Compare
             return m_collator.compare(left, right);
         }
 
+        int operator()(const Path &left, const Path &right) const
+        {
+            return m_collator.compare(left.toString(), right.toString());
```

Comment:
Since this looks like a shim, I would write it:
```c++
return operator()(left.toString(), right.toString());
```
The same applies to other new functions in this file.

Reply:
On a second thought, adding to `Utils::Compare` might be a bad idea.
IMO there will be 2 types of comparison for a path type. The first is actual path comparison where the case sensitivity of the file system (or perhaps operating system) will be taken into consideration. The second is comparing the path on a string level where the case sensitivity is always true when comparing.
And this is not clear which kind of comparison `Utils::Compare` is doing. For string like comparison, I would just convert the path to string at the usage site instead.

Reply:
If scope of this class is limited only to string comparison it should probably be moved in `utils/string/compare.h` as `Utils::String::Compare`.

Reply:
>For string like comparison, I would just convert the path to string at the usage site instead.

If I remember correctly it is used in some kind of sort algorithms so I will need to implement another Compare for Path class and just duplicate all the logic from existing Compare to compare paths in the same way as it was done previously... Have you any better idea?

Reply:
Seems it is used in only one place so I ended up with removing this operators from Compare, and now I process paths as strings there.